### DISCUSSION TITLE
ImPlot Very Long Compile Time

### DISCUSSION THREAD

**Discussion by rishineedscookies** [epezent/implot#338] [Q&A]
I'd love to use ImPlot, but unfortunately it completely bloats my compile times from a few seconds to >30 seconds. I'm not entirely sure what the root of the long compile times is since there's only a few thousand lines of code which should normally compile near instantly on a modern computer. I've tried using incremental linking, unity builds, multithreaded compiles, but they are all very slow because of ImPlot. Is there a plan or direction to try to improve the compile times?

**Comment by epezent** [comment]
Thanks for raising this issue. It's something I'm aware of and realize has gotten out of hand. The trouble maker is `implot_items.cpp`, which is template and inline heavy. You can speed up compilation of this file by defining `IMPLOT_NO_FORCE_INLINE`. Past testing suggests extreme inlining provides a noticable performance boost to ImPlot, but it does unfortunately bloat the compile time. I have some work in progress that may improve this a bit, and I've considered omitting some of the supported data types by default to cut down compile time/size even further (see #328).
